server/controllers/api/claim/publish/validateFileForPublish.js

```diff
@@ -0,0 +1,80 @@
+const logger = require('winston');
+
+const { publishing } = require('@config/siteConfig.json');
+
+const {
+  fileSizeLimits: {
+    image: maxSizeImage = 10000000,
+    video: maxSizeVideo = 50000000,
+    audio: maxSizeAudio = 50000000,
+    text: maxSizeText = 50000000,
+    model: maxSizeModel = 50000000,
+    application: maxSizeApplication = 50000000,
+    customByContentType = {},
+  },
+} = publishing;
+
+const SIZE_MB = 1000000;
+
+const validateFileForPublish = file => {
+  let contentType = file.type;
+  let mediaType = contentType ? contentType.substr(0, contentType.indexOf('/')) : '';
+  console.log(file);
+
+  if (!file) {
+    throw new Error('no file provided');
+  }
+
+  if (/'/.test(file.name)) {
+    throw new Error('apostrophes are not allowed in the file name');
+  }
+
+  if (Object.keys(customByContentType).includes(contentType)) {
+    if (file.size > customByContentType[contentType]) {
+      throw new Error(
+        `Sorry, type ${contentType} is limited to ${customByContentType[contentType] / SIZE_MB} MB.`
+      );
+    }
+  } else {
+    switch (mediaType) {
```

Comment:
Can make an object of `{ [mediaType]: maxSize }`, which would map to:
```
let mediaTypeLimit = mediaTypeLimits[mediaType] || false;
if (mediaTypeLimit) {
  if (file.size > mediaTypeLimit) {
    throw new Error(`Sorry, type ${mediaType} is limited to ${mediaTypeLimit / SIZE_MB} MB.`);
  }
}
```

Reply:
Done and DRY.